lib/src/classifier/knn_classifier/knn_classifier_impl.dart

```diff
@@ -0,0 +1,150 @@
+import 'package:ml_algo/src/classifier/knn_classifier/knn_classifier.dart';
+import 'package:ml_algo/src/helpers/validate_test_features.dart';
+import 'package:ml_algo/src/knn_kernel/kernel.dart';
+import 'package:ml_algo/src/knn_solver/knn_solver.dart';
+import 'package:ml_algo/src/knn_solver/neigbour.dart';
+import 'package:ml_algo/src/predictor/assessable_predictor_mixin.dart';
+import 'package:ml_dataframe/ml_dataframe.dart';
+import 'package:ml_linalg/dtype.dart';
+import 'package:ml_linalg/matrix.dart';
+import 'package:ml_linalg/vector.dart';
+
+class KnnClassifierImpl with AssessablePredictorMixin implements KnnClassifier {
+  KnnClassifierImpl(
+      String targetName,
+      this._classLabels,
+      this._kernel,
+      this._solver,
+      this._dtype,
+  ) : classNames = [targetName] {
+    if (_classLabels.isEmpty) {
+      throw Exception('Empty class label list provided');
+    }
+  }
+
+  @override
+  final List<String> classNames;
+
+  final List<num> _classLabels;
+  final Kernel _kernel;
+  final KnnSolver _solver;
+  final DType _dtype;
+
+  @override
+  DataFrame predict(DataFrame features) {
+    validateTestFeatures(features, _dtype);
+
+    final labelsToProbabilities = _getLabelsToProbabilitiesMapping(features);
+    final labels = labelsToProbabilities.keys.toList();
+    final predictedOutcomes = _getProbabilityMatrix(labelsToProbabilities)
+        .rows
+        .map((row) => labels[row.toList().indexOf(row.max())])
+        .toList();
+
+    final outcomesAsVector = Vector.fromList(predictedOutcomes, dtype: _dtype);
+
+    return DataFrame.fromMatrix(
+      Matrix.fromColumns([outcomesAsVector], dtype: _dtype),
+      header: classNames,
+    );
+  }
+
+  @override
+  DataFrame predictProbabilities(DataFrame features) {
+    final labelsToProbabilities = _getLabelsToProbabilitiesMapping(features);
+    final probabilityMatrix = _getProbabilityMatrix(labelsToProbabilities);
+
+    final header = labelsToProbabilities
+        .keys
+        .map((label) => label.toString());
```

Comment:
consider postfix for class label in the header, add unit test for `predictProbabilities`